Doc/library/socket.rst

```diff
@@ -595,6 +595,24 @@ The following functions all create :ref:`socket objects <socket-objects>`.
    .. versionchanged:: 3.2
       *source_address* was added.
 
+.. function:: bind_socket(address, family=AF_UNSPEC, type=SOCK_STREAM, *,
+                          backlog=128, reuse_addr=None, reuse_port=False,
+                          flags=AI_PASSIVE):
+
+    Convenience function which creates a socket bound to *address* (a 2-tuple
+    ``(host, port)``) and return the socket object upon which you can call
+    :meth:`socket.accept()` in order to accept new connections.
+    If *host* is an empty string or ``None`` all network interfaces are assumed.
+    If *family* is :data:`AF_UNSPEC` address family will be determined from
+    the *host* specified in *address*.
+    *type* should be either :data:`SOCK_STREAM` or :data:`SOCK_DGRAM`.
+    *backlog* is the queue size passed to :meth:`socket.listen` if
+    :data:`SOCK_STREAM` *type* is used.
+    *reuse_addr* and *reuse_port* dictates whether to use :data:`SO_REUSEADDR`
+    and :data:`SO_REUSEPORT` socket options.
```

Comment:
What does _reuse_addr_=None mean? It looks like it enables REUSEADDR if supported, but doesn’t enforce it.

Reply:
Correct. One may use `reuse_addr_=False` to explicitly disable the option, but if left unspecified (``None), what you want 99% of the times is to use SO_REUSEADDR. I took inspiration from `asyncio` which does the same.